ENG-7705: Add Firebase 11.x support to analytics-ios-integration-firebase

tc opened the pull request:
## 📍 Ticket
- ENG-7705 (Firebase SDK Update Support)

## 🆕 What's new?
Updated Firebase dependency constraints in the Segment-Firebase integration to support Firebase 11.x, resolving dependency conflicts that prevented the Speak iOS app from updating to Firebase 11.14.0.

### Changes Made:
- **Updated Firebase dependency constraints** in `Segment-Firebase.podspec`:
  - Changed from `~> 10.6.0` (which excludes 11.x) to `>= 10.23` (which allows 11.x)
  - Applied to Firebase, Firebase/Core, and FirebaseAnalytics dependencies
- **Bumped version** from 2.7.13 to 1.1.0 to indicate Firebase 11.x compatibility
- **Removed Package.resolved** to allow fresh dependency resolution with new constraints

### Compatibility Notes:
- Maintains backward compatibility with Firebase 10.23+
- Enables Firebase 11.x support for dependent projects
- Resolves dependency conflict: `'analytics-ios-integration-firebase' 1.0.0 depends on 'firebase-ios-sdk' 10.23.0..<11.0.0`

## 👨‍🏫 How to test?
- Verify package builds successfully with Firebase 11.x
- Test Segment-Firebase integration functionality remains intact
- Confirm dependency resolution works in consuming projects (like Speak iOS)

## 📸 Can I see?
This is a dependency constraint update with no functional changes. The integration behavior remains identical while supporting newer Firebase versions.

---

**Related PR**: usespeakeasy/speak-ios#2937 (Firebase 11.14.0 update)
**Link to Devin run**: https://app.devin.ai/sessions/33f13994ea4b48adb73db1144fa67b64

devin-ai-integration[bot] committed:
Update Firebase dependency constraints to support Firebase 11.x

- Updated podspec Firebase dependencies from '~> 10.23' to '>= 10.23' to allow Firebase 11.x
- Bumped version to 1.1.0 to indicate compatibility with Firebase 11.x
- Removed Package.resolved to allow fresh dependency resolution
- Maintains backward compatibility with Firebase 10.23+

Resolves dependency conflict preventing Firebase 11.14.0 usage in Speak iOS app

Co-Authored-By: tommy@usespeak.com <tommy.chheng@gmail.com>

tc commented on Segment-Firebase.podspec:
```diff
@@ -1,6 +1,6 @@
 Pod::Spec.new do |s|
   s.name             = "Segment-Firebase"
-  s.version          = "2.8.0"
+  s.version          = "1.1.0"
```
change to 2.9.0 vs 1.1.0

tc commented:
@salva-dor feel free to merge if you approve.

 since we bump the version, this wouldn't break anything?

devin-ai-integration[bot] committed:
Update version to 2.9.0 following semantic versioning

- Address GitHub PR feedback from tc
- Follow semantic versioning from previous 2.7.13 to 2.9.0
- Maintains Firebase 11.x compatibility support

Co-Authored-By: tommy@usespeak.com <tommy.chheng@gmail.com>

salva-dor commented:
Devin, this repository has been modified to support SPM. Please update the Package.swift file instead of any CocoaPods-related files.

salva-dor commented:
aside: @sberrevoets looks like you updated already the package, was that related to this ticket? https://linear.app/speak/issue/ENG-7705/update-ga4f-firebase-sdk-to-version-11140-on-ios

devin-ai-integration[bot] committed:
Update Firebase dependency to support Firebase 11.x in Package.swift for SPM

- Changed Firebase dependency from 'from: 10.23.0' to 'from: 11.0.0'
- Addresses GitHub comment to use Package.swift instead of podspec for SPM support
- Ensures Firebase 11.14.0 compatibility for dependent projects

Related: ENG-7705

Co-Authored-By: tommy@usespeak.com <tommy.chheng@gmail.com>

sberrevoets commented:
@salva-dor No I didn't know that ticket existed! There was just a renovate job that didn't pass so I fixed it. Firebase should already be at 11.x in prod as of earlier this week.

tc commented:
> @salva-dor No I didn't know that ticket existed! There was just a renovate job that didn't pass so I fixed it. Firebase should already be at 11.x in prod as of earlier this week.

oh great, so we don't need this PR?

glad renovate did our job!

salva-dor commented:
Renovate + @sberrevoets 😅, so we can close this PR!
